mslice/models/slice/matplotlib_slice_plotter.py

```diff
@@ -40,6 +44,77 @@ def _getDisplayName(self, axisUnits, comment=None):
         else:
             return axisUnits
 
+    def _show_plot(self, workspace_name, plot_data, extent, colourmap, norm, x_axis, y_axis):
+        plt.imshow(plot_data, extent=extent, cmap=colourmap, aspect='auto', norm=norm,
+                   interpolation='none', hold=False)
+        plt.title(workspace_name)
+        comment = self._slice_algorithm.getComment(str(workspace_name))
+        plt.xlabel(self._getDisplayName(x_axis.units, comment))
+        plt.ylabel(self._getDisplayName(y_axis.units, comment))
+        plt.gcf().get_axes()[1].set_ylabel('Intensity (arb. units)', labelpad=20, rotation=270)
+
+    def show_scattering_function(self, workspace):
+        slice_cache = self.slice_cache[workspace]
+        self._show_plot(workspace, slice_cache['plot_data'][0], slice_cache['boundaries'], slice_cache['colourmap'],
+                        slice_cache['norm'], slice_cache['x_axis'], slice_cache['y_axis'])
+
+    def show_dynamical_susceptibility(self, workspace):
+        slice_cache = self.slice_cache[workspace]
+        if slice_cache['plot_data'][1] is None:
+            raise ValueError('plot_data not calculated')
+        self._show_plot(workspace, slice_cache['plot_data'][1], slice_cache['boundaries'], slice_cache['colourmap'],
+                        slice_cache['norm'], slice_cache['x_axis'], slice_cache['y_axis'])
+
+    def show_dynamical_susceptibility_magnetic(self, workspace):
+        slice_cache = self.slice_cache[workspace]
+        if slice_cache['plot_data'][1] is None:
+            raise ValueError('plot_data not calculated')
+        self._show_plot(workspace, slice_cache['plot_data'][2], slice_cache['boundaries'], slice_cache['colourmap'],
+                        slice_cache['norm'], slice_cache['x_axis'], slice_cache['y_axis'])
+        plt.gcf().get_axes()[1].set_ylabel('chi\'\'(Q,E) |F(Q)|$^2$ ($mu_B$ $meV^{-1} sr^{-1} f.u.^{-1}$)',
+                                           rotation=270, labelpad=20)
+
+    def add_sample_temperature_field(self, field_name):
+        self._sample_temp_fields.append(field_name)
+
+    def update_sample_temperature(self, workspace):
+        temp = self.sample_temperature(str(workspace))
+        self.slice_cache[workspace]['sample_temp'] = temp
+        self.slice_cache[workspace]['plot_data'][1] = self._slice_algorithm.compute_chi(
+            self.slice_cache[workspace]['plot_data'][0], temp, self.slice_cache[workspace]['y_axis'])
+        self.slice_cache[workspace]['plot_data'][2] = self._slice_algorithm.compute_chi_magnetic(
+            self.slice_cache[workspace]['plot_data'][1])
+
+    def sample_temperature(self, ws_name):
```

Comment:
I think this method should move to the `mantid_slice_algorithm` class and then the `AnalysisDataService` dependency can be removed from here.

Can we also add a comment that we have to go back to the unprojected workspace because mantid drops the log information on the way from the reduced data to the projection when going via `SofQW`.